osgar/obstdet3d.py

```diff
@@ -0,0 +1,25 @@
+"""
+  Obstacle Detection 3D
+"""
+import numpy as np
+
+from osgar.node import Node
+
+
+class ObstacleDetector3D(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register('obstacle')
+
+    def on_depth(self, data):
+        assert data.shape == (400, 640), data.shape
+        selection = data[150:250, 300:340]
+        mask = selection > 0  # not valid data?
+        if mask.max() == True:
+            dist = selection[mask].min() / 1000
+        else:
+            dist = 0.0
+        self.publish('obstacle', float(dist))
```

Comment:
Is the float necessary?

Reply:
float is needed in order not to propagate numpy structure. Or if you mean why not to use millimeters and int ... then yes, I would like fractional numbers more :)